### ISSUE TITLE
set_requires_grad

### ISSUE THREAD

**Issue by Xudong7Chen** [junyanz/pytorch-CycleGAN-and-pix2pix#1211]
When training the GAN, if i delete the following code lines
set_requires_grad(net_D, True)
and,
set_requires_grad(net_D, False)
Will this affect the final model training performance?

Looking forward to your answer.
Thank you.

**Comment by JunlinHan** [comment]
Yes. It definitely affects the training.

**Comment by Xudong7Chen** [comment]
> Yes. It definitely affects the training.

Thanks for you reply.
In another issues, authors say that it's a small speedup trick. set_requires_grad=False will stop calculating gradients for the discriminator while we update the generator. It can save some time and memory. Generators will still get the gradients.

Will set_requires_grad=False only affect the training efficiency, or it will also affect the final trained model accuracy?

**Comment by JunlinHan** [comment]
> 
> 
> > Yes. It definitely affects the training.
> 
> Thanks for you reply.
> In another issues, authors say that it's a small speedup trick. set_requires_grad=False will stop calculating gradients for the discriminator while we update the generator. It can save some time and memory. Generators will still get the gradients.
> 
> Will set_requires_grad=False only affect the training efficiency, or it will also affect the final trained model accuracy?

I think it won't hurt the performance if you calculate the gradient of discriminator when updating generator. So it oonly affect the training efficiency.

**Comment by Xudong7Chen** [comment]
> > > Yes. It definitely affects the training.
> > 
> > 
> > Thanks for you reply.
> > In another issues, authors say that it's a small speedup trick. set_requires_grad=False will stop calculating gradients for the discriminator while we update the generator. It can save some time and memory. Generators will still get the gradients.
> > Will set_requires_grad=False only affect the training efficiency, or it will also affect the final trained model accuracy?
> 
> I think it won't hurt the performance if you calculate the gradient of discriminator when updating generator. So it only affect the training efficiency.

Thanks for your reply.
